src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -84,22 +92,22 @@
     /** Period literal. */
     private static final String PERIOD = ".";
 
-    /**
-     * Stores allowed values in document for inherit doc literal.
-     */
-    private static final Set<Integer> SKIP_TOKENS = new HashSet<>(
-        Arrays.asList(JavadocTokenTypes.NEWLINE,
-                       JavadocTokenTypes.LEADING_ASTERISK,
-                       JavadocTokenTypes.EOF)
+    /** Inherit doc literal. */
+    private static final String INHERIT_DOC = "{@inheritDoc}";
```

Comment:
@sagar-shah94 Why don't you want to match against token? it is the most precise way to find a specific thing like the inherit doc literal. Match text is slower and not as accurate.
Some tokens like descriptions could have same text as inherit doc but not be true inherit docs.
We use parsers for a reason, you should rely on them more.